go/database/mpt/forest_test.go

```diff
@@ -509,27 +567,36 @@ func TestForest_getAccess_Fails(t *testing.T) {
 					if err != nil {
 						t.Fatalf("failed to open forest: %v", err)
 					}
+					defer forest.Close()
 
 					// inject failing stock to trigger an error applying the update
 					var injectedErr = errors.New("failed to call Get")
 					ctrl := gomock.NewController(t)
 					cache := NewMockNodeCache(ctrl)
 					cache.EXPECT().Get(gomock.Any()).AnyTimes().Return(nil, false)
 					var n Node
+					// only the second call must fail - repeats four times for four calls
 					cache.EXPECT().GetOrSet(gomock.Any(), gomock.Any()).AnyTimes().Return(shared.MakeShared(n), false, EmptyId(), nil, false)
 					cache.EXPECT().Touch(gomock.Any()).AnyTimes()
-					forest.nodeCache = cache
 
```

Comment:
Removing the mocked node cache here makes the expectation setup above unnecessary. Right now it is not failing, since all expectations are set as `AnyTimes`.

Should this cache be mocked? If so, this line needs to be re-added and most likely a backup of the original cache restored.

Reply:
Right, my bad.